src/osd/PrimaryLogPG.cc

```diff
@@ -1565,6 +1565,15 @@ void PrimaryLogPG::calc_trim_to()
     pg_trim_to = new_trim_to;
     assert(pg_trim_to <= pg_log.get_head());
     assert(pg_trim_to <= min_last_complete_ondisk);
+  } else if (pg_trim_to == eversion_t()) {
```

Comment:
IIUC, in this case, the pg just finished peering. but the primary's pg log size is less than the target trimming size, so we simply skip the trim in this round. but with your change, we will trim to the `tail` of primary's pg log and the upper bound is also clamped by the `limit`.

i think we can go further by trimming some entries in primary's pg log as long as we don't exceed the the `limit`.

the flip side of this change is, we will **always** trim the pg_log after peering, even if the number of entries to trim is less than `osd_min_pg_log_entries`. but considering the peering process should not happen very frequent in a healthy cluster. so we should be okay with it.